posts/2021-02-11-Rust-1.50.0.md

```diff
@@ -0,0 +1,140 @@
+---
+layout: post
+title: "Announcing Rust 1.50.0"
+author: The Rust Release Team
+release: true
+---
+
+The Rust team is happy to announce a new version of Rust, 1.50.0. Rust is a
+programming language that is empowering everyone to build reliable and
+efficient software.
+
+If you have a previous version of Rust installed via rustup, getting Rust
+1.50.0 is as easy as:
+
+```console
+rustup update stable
+```
+
+If you don't have it already, you can [get `rustup`][install]
+from the appropriate page on our website, and check out the
+[detailed release notes for 1.50.0][notes] on GitHub.
+
+[install]: https://www.rust-lang.org/install.html
+[notes]: https://github.com/rust-lang/rust/blob/master/RELEASES.md#version-1500-2021-02-11
+
+## What's in 1.50.0 stable
+
+For this release, we have improved array indexing, expanded union field safety, refined file descriptors, and added to the standard library.
+See the [detailed release notes][notes] to learn about other changes
+not covered by this post.
+
+### Const-generic array indexing
+
+Continuing the march toward stable `const` generics, this release adds
+implementations of `ops::Index` and `IndexMut` for arrays `[T; N]` for
+_any_ length of `const N`. The indexing operator `[]` already worked on
+arrays through built-in compiler magic, but at the type level, arrays
+didn't actually implement the library traits until now.
+
+```rust
+fn second<C>(container: &C) -> &C::Output
+where
+    C: std::ops::Index<usize> + ?Sized,
+{
+    &container[1]
+}
+
+fn main() {
+    let array: [i32; 3] = [1, 2, 3];
+    assert_eq!(second(&array[..]), &2); // slices worked before
+    assert_eq!(second(&array), &2); // now it also works directly
+}
+```
+
+### Safe assignments to `ManuallyDrop<T>` union fields
+
+Rust 1.49 made it possible to add `ManuallyDrop<T>` fields to a `union`
+as part of allowing `Drop` for unions at all; however, unions don't know
+which variant is valid (they don't drop any old value on assignment), so
+safe Rust has limited this to `Copy` types only, which never `Drop`. Of
+course, that's also true of `ManuallyDrop`, so now Rust 1.50 allows safe
+assignments to these fields as well.
```

Comment:
Can we split the first sentence of this paragraph up? I think it's carrying a lot of weight right now in a relatively short space and would be nice to add some breaks for readers.

Reply:
I split and rearranged this a bit.